extensions/azure-functions-resteasy/maven-archetype/pom.xml

```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2018 Red Hat, Inc.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <artifactId>quarkus-azure-functions-resteasy-parent</artifactId>
+    <groupId>io.quarkus</groupId>
+    <version>999-SNAPSHOT</version>
+    <relativePath>../</relativePath>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>quarkus-azure-functions-resteasy-archetype</artifactId>
+  <packaging>maven-archetype</packaging>
```

Comment:
I don't think we want a maven archetype, we are not using them for anything else. This should be part of our CLI tooling instead IMHO.

Reply:
I urge you to look at the archetype.   I think we're stuck with it for now until improvements can be made.  Even then, I think the archetype might be a better approach:

Azure deployment descriptors are required.  Descriptors that the user may want to edit between builds.  A specific directory structure is required that needs a copy of an uberjar in a specific directory location.  Quarkus extensions can not bind to a post-build phase, don't have knowledge of the target/ directory,  and they do not have access to pom properties.    The archetype also pulls in the azure-maven-plugin so the generated deployment can be pushed to Azure.  I don't want to be maintaining a fork of that.  Finally, I don't see why this would be moved to a CLI tool. Bundling a deployment is purely a build action.

Reply:
One more thing, as I posted in a private message, I also don't think its possible to have this azure extension in one pom with the core code of the app.  Dev/test mode needs to load different extensions and I don't think the Test classloader pulls in transitive dependencies correctly.  Combine this with all the above issues: I think an archetype is the best solution at the moment.